Source/Diagnostics/FlushFormats/FlushFormatOpenPMD.cpp

```diff
@@ -12,13 +12,35 @@ FlushFormatOpenPMD::FlushFormatOpenPMD (const std::string& diag_name)
     // Which backend to use (ADIOS, ADIOS2 or HDF5). Default depends on what is available
     std::string openpmd_backend {"default"};
     // one file per timestep (or one file for all steps)
-    bool openpmd_tspf = true;
+    std::string  openpmd_encoding {"v"};
     pp_diag_name.query("openpmd_backend", openpmd_backend);
-    pp_diag_name.query("openpmd_tspf", openpmd_tspf);
-    auto & warpx = WarpX::GetInstance();
-    m_OpenPMDPlotWriter = std::make_unique<WarpXOpenPMDPlot>(
-        openpmd_tspf, openpmd_backend, warpx.getPMLdirections()
-    );
+    bool encodingDefined = pp_diag_name.query("openpmd_encoding", openpmd_encoding);
```

Comment:
At this point, also query `<diag>.diag_type` and check for `BackTransformed` - if that is the case, make the default `g` please, so that users get the right thing default if they don't specify something.

Reply:
Oh, I didn't know that.  Will add query <diag_type>

Reply:
Yes, that would be practical I think for user-friendly defaults :)
query via `pp_diag_name.query("diag_type", diag_type);`